src/RobotInterface/YarpImplementation/include/BipedalLocomotion/RobotInterface/YarpSensorBridgeImpl.h

```diff
@@ -1318,6 +1556,234 @@ struct YarpSensorBridge::Impl
+
+        return true;
+    }
+
+    /**
+     * Read motor encoders measurements
+     */
+    bool readAllMotorEncoders(bool checkForNan = false)
+    {
+        if (!metaData.bridgeOptions.isMotorEncodersEnabled || !metaData.bridgeOptions.isKinematicsEnabled)
+        {
+            // do nothing
+            return true;
+        }
+
+        constexpr auto logPrefix = "[YarpSensorBridge::Impl::readAllMotorEncoders]";
+        bool ok{ true };
+        ok = ok && controlBoardRemapperInterfaces.motorEncoders->getMotorEncoders(
+                 controlBoardRemapperMeasures.motorPositionsUnordered.data());
+        ok = ok && controlBoardRemapperInterfaces.motorEncoders->getMotorEncoderSpeeds(
+                 controlBoardRemapperMeasures.motorVelocitiesUnordered.data());
+
+        if (!ok)
+        {
+            log()->error("{} Unable to read from motor encoders, use previous measurement.",
+                         logPrefix);
+            return false;
+        }
+
+        if (checkForNan)
+        {
+            if (nanExistsInVec(controlBoardRemapperMeasures.motorCurrentsUnordered,
+                               logPrefix,
+                               "MotorEncoders"))
+            {
+                return false;
+            }
+        }
+
+        controlBoardRemapperMeasures.motorPositions.noalias()
+            = controlBoardRemapperMeasures.remappedJointPermutationMatrix
+              * controlBoardRemapperMeasures.motorPositionsUnordered;
+        controlBoardRemapperMeasures.motorVelocities.noalias()
+            = controlBoardRemapperMeasures.remappedJointPermutationMatrix
+              * controlBoardRemapperMeasures.motorVelocitiesUnordered;
+
+        controlBoardRemapperMeasures.receivedTimeInSeconds = yarp::os::Time::now();
+
+        return true;
+    }
+
+    /**
+     * Read motor encoders measurements
+     */
+    bool readAllMotorPWMs(bool checkForNan = false)
+    {
+        if (!metaData.bridgeOptions.isPWMControlEnabled || !metaData.bridgeOptions.isKinematicsEnabled)
+        {
+            // do nothing
+            return true;
+        }
+
+        constexpr auto logPrefix = "[YarpSensorBridge::Impl::readAllMotorPWMs]";
+        bool ok{ true };
+        for (int j = 0; j < controlBoardRemapperMeasures.motorPWMsUnordered.size(); j++)
+        {
+            ok &= controlBoardRemapperInterfaces.amp->getPWM(j, &(controlBoardRemapperMeasures.motorPWMsUnordered[j]));
+        }
+
+        if (!ok)
+        {
+            log()->error("{} Unable to read from amplifiers, use previous measurement.",
+                         logPrefix);
+            return false;
+        }
+
+        if (checkForNan)
+        {
+            if (nanExistsInVec(controlBoardRemapperMeasures.motorPWMsUnordered,
+                               logPrefix,
+                               "MotorPWMs"))
+            {
+                return false;
+            }
+        }
+
+        controlBoardRemapperMeasures.motorPWMs.noalias()
```

Comment:
Is this valid also for coupled joints?

Reply:
Actually I'm not sure about it. I saw something similar in 
https://github.com/robotology/yarp-telemetry/blob/7e2e551bbd1de58cd048c4aab6d4cf8b422a3b13/src/telemetryDeviceDumper/TelemetryDeviceDumper.cpp#L569-L572
but they call it `jointPWM` instead of `motorPWM`.
The documentation is not very clear to me https://github.com/robotology/yarp/blob/2a0eacdf9b2e6f9be04cd2ebcbe097208b758838/src/libYARP_dev/src/yarp/dev/IAmplifierControl.h#L142-L148